datamap/mv/plan/src/main/scala/org/apache/carbondata/mv/plans/modular/ModularRelation.scala

```diff
@@ -47,14 +47,7 @@ case class ModularRelation(databaseName: String,
   override def computeStats(spark: SparkSession, conf: SQLConf): Statistics = {
     val plan = spark.table(s"${ databaseName }.${ tableName }").queryExecution.optimizedPlan
     val stats = SparkSQLUtil.invokeStatsMethod(plan, conf)
-    val output = outputList.map(_.toAttribute)
-    val mapSeq = plan.collect { case n: logical.LeafNode => n }.map {
-      table => AttributeMap(table.output.zip(output))
-    }
-    val rewrites = mapSeq(0)
-    val attributeStats = AttributeMap(stats.attributeStats.iterator
-      .map { pair => (rewrites(pair._1), pair._2) }.toSeq)
-    Statistics(stats.sizeInBytes, stats.rowCount, attributeStats, stats.hints)
+    SparkSQLUtil.getStatisticsObj(outputList, plan, stats)
```

Comment:
Its not a good idea to keep most of the code in reflections. so extract a method like a below and move them to spark 2.1 and 2.2 packages .

```
private def getStatistics(plan: LogicalPlan,
      stats: Statistics,
      output: Seq[Attribute]) = {
    val mapSeq = plan.collect { case n: logical.LeafNode => n }.map {
      table => AttributeMap(table.output.zip(output))
    }
    val rewrites = mapSeq(0)
    val attributeStats = AttributeMap(stats.colStats.iterator
      .map { pair => (rewrites(pair._1), pair._2) }.toSeq)
    Statistics(stats.sizeInBytes, stats.rowCount, attributeStats, stats.hints)
  }
```

Reply:
Do we add the same code to spark 2.3 CarbonToSparkAdapater.scala? or we add a common   CarbonToSparkAdapater.scala to "commonTo2.2And2.3"?

Most of this code is different between spark2.1 and spark 2.2 & 2.3,  so i think using reflections is OK，can we keep this code??

Reply:
ok